RFS/src/main/java/com/rfs/RunRfsWorker.java

```diff
@@ -152,6 +163,13 @@ public static void main(String[] args) throws Exception {
             IndexCreator_OS_2_11 indexCreator = new IndexCreator_OS_2_11(targetClient);
             IndexRunner indexWorker = new IndexRunner(globalState, cmsClient, snapshotName, indexMetadataFactory, indexCreator, transformer);
             indexWorker.run();
+
+            ShardMetadata.Factory shardMetadataFactory = new ShardMetadataFactory_ES_7_10(repoDataProvider);
```

Comment:
Can you look into how SimpleRestoreFromSnapshot_ES_7_10 can be updated to use this code instead of the ad-hoc version?  Seems like a great opportunity to reuse this system

Reply:
100%, agreed.  Will try to work into one of the small cleanup tasks I have next.

Reply:
Tried to get this started, but seems like a refactor might be better to get this working: https://github.com/chelma/opensearch-migrations/compare/main...peternied:documents-runner-test